lib/src/backend.rs

```diff
@@ -519,4 +521,14 @@ pub trait Backend: Send + Sync + Debug {
     /// timestamps may have less precision than the millisecond precision in
     /// `Commit`.
     fn write_commit(&self, contents: Commit) -> BackendResult<(CommitId, Commit)>;
+
+    /// Identical to `write_commit`, except the signing function is called with
+    /// some binary representation of the commit before it is created to
+    /// generate a cryptographic (hopefully) signature which will be stored in
+    /// the commit metadata.
+    fn write_commit_signed(
+        &self,
+        contents: Commit,
+        signing_fn: SigningFn,
+    ) -> BackendResult<(CommitId, Commit)>;
```

Comment:
Actually, it might be okay to extend `write_commit()` to accept `Option<SignedFn>` instead of adding two separate functions.

`Backend` functions are low level, and proxied through the `Store`. I think a convenient API can be added to `Store` or `CommitBuilder`.

Reply:
Ok, done
It's actually way less passing `None` around than I thought, so it's great